### PR TITLE
dolphin-emu: 5.0-21460 -> 2407, adopt

### PR DESCRIPTION
**Pull request by pbsds**
Diff: https://github.com/dolphin-emu/dolphin/compare/a9544510468740b77cf06ef28daaa65fe247fd32...b92e354389bb7c0bd114a8631b8af110d3cb3a14

Announcement: https://dolphin-emu.org/blog/2024/07/02/dolphin-releases-announcement/

It's sad to see so many great contributors leave. Too bad the previous dolphin-emu maintainers barely didn't make it to this big release.

I'll adopt dolpin-emu to remind myself to play games more often, and I could really use a darwin co-maintainer. :pray:

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by AndersonTorres** [comment]
I was going to adopt this, indeed my intention is to somehow group such emulators since they use similar technologies.

**Comment by wegank** [comment]
`dolphin-emu.passthru.tests` failed on darwin, it looks like `Dolphin 5.0` should be `Dolphin 2407.0`.

File: pkgs/applications/emulators/dolphin-emu/default.nix [hunk 1]
```diff
@@ -58,21 +58,16 @@
...
   pname = "dolphin-emu";
-  version = "5.0-21460";
+  version = "2407";
 
   src = fetchFromGitHub {
     owner = "dolphin-emu";
     repo = "dolphin";
-    rev = "a9544510468740b77cf06ef28daaa65fe247fd32";
-    hash = "sha256-mhD7Uaqi8GzHdR7Y81TspvCnrZH2evWuWFgXMQ2c8g0=";
+    rev = "b92e354389bb7c0bd114a8631b8af110d3cb3a14";
```
Comment (wegank) [inline pkgs/applications/emulators/dolphin-emu/default.nix:66]:
```suggestion
    rev = "refs/tags/${version}";
```

Comment (pbsds) [reply pkgs/applications/emulators/dolphin-emu/default.nix:66]: I opted to hold of on that until we were sure that they publish following releases as tags. But i found https://github.com/dolphin-emu/central/blob/1960442b2902959ebb39f536b7cab701ef189dce/central/buildbot.py#L61 which indicate they likely will.

**Review by qubitnano — APPROVED** [review]
Result of `nixpkgs-review pr 324187` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dolphin-emu</li>
  </ul>
</details>

**Pull request merged**